browser/src/helpers/blobs.tsx

```diff
@@ -1,9 +1,70 @@
-import React from "react";
-import { Pattern, Prompt } from "../types/common/server-api";
-
-export function renderBlob(
-  pattern: Pattern,
-  prompt: Prompt
-): React.ReactElement {
-  return <></>;
+import React, {useCallback, useEffect, useMemo, useState} from "react";
+import {Contribution, Prompt} from "../types/common/server-api";
+import { sha256 } from "ethers/lib/utils";
+import Blob from "../components/Blob";
+import { OrbitControls } from "@react-three/drei";
+import { Canvas } from "@react-three/fiber";
+import {Placeholder} from "../components/ContributionSection";
+
+function toHex(str: string) {
+  return str
+    .split("")
+    .map((_, i) => str.charCodeAt(i).toString(16))
+    .join("");
+}
+
+function fetch(message: string, idx: number, val_start: number, range: number) {
```

Comment:
why is this called `fetch` ? makes me think it's going to do an ajax call lol

Reply:
yeah my bad LOL was not thinking when I was naming